src/quota.rs

```diff
@@ -0,0 +1,201 @@
+    },
+};
+
+/// Generates a detailed report about the current Quota usage on the for deltachat relevant folders
+/// and sends it to the user via [add_device_msg]
+///
+/// It's a bit like the prepaid mobile carrier service menu/messages,
+/// where you type a special number and then get a message back with your current balance.
+pub(crate) async fn quota_usage_report_job(context: &Context, imap: &mut Imap) -> Result<()> {
+    if !imap.capabilities_determined() {
+        bail!("imap is not ready");
+    }
+
+    if imap.can_check_quota() {
+        let folders = get_watched_folders(context).await;
+        let unique_quota_roots = get_unique_quota_roots_and_usage(folders, imap).await?;
+
+        // build and send report message
+        let mut msg = Message::new(Viewtype::Text);
+        msg.text = Some(generate_report_message(&unique_quota_roots, context).await?);
+        add_device_msg(context, None, Some(&mut msg)).await?;
+    } else {
+        warn!(
+            context,
+            "the email server does not support the quota extention"
+        );
+        let mut msg = Message::new(Viewtype::Text);
+        msg.text = Some(quota_not_supported(context).await); // todo stock string?
+        add_device_msg_with_importance(context, None, Some(&mut msg), true).await?;
+    }
+
+    Ok(())
+}
+
+async fn generate_report_message(
+    unique_quota_roots: &[(String, Vec<QuotaResource>)],
+    context: &Context,
+) -> Result<String> {
+    let mut message: String = "".to_owned();
+
+    let storage_stock_string = quota_resource_storage(context).await;
+    let messages_stock_string = quota_resource_messages(context).await;
+    for (name, quota_resources) in unique_quota_roots {
+        message.push_str(&format!("{}:\n", &name));
+        use async_imap::types::QuotaResourceName::*;
+        for resource in quota_resources {
+            message.push_str(&match &resource.name {
+                Atom(name) => {
+                    format!("[{}/{}] {}\n", resource.usage, resource.limit, name)
+                }
+                Message => {
+                    format!(
+                        "{}/{} {}\n",
+                        resource.usage, resource.limit, messages_stock_string
+                    )
+                }
+                Storage => {
+                    let used = (resource.usage * 1024)
+                        .file_size(file_size_opts::BINARY)
+                        .map_err(|err| anyhow!("{}", err))?;
+                    let limit = (resource.limit * 1024)
+                        .file_size(file_size_opts::BINARY)
+                        .map_err(|err| anyhow!("{}", err))?;
+                    format!("{}/{} {}\n", used, limit, storage_stock_string)
+                }
+            });
+        }
+    }
+    Ok(message)
+}
+
+async fn get_unique_quota_roots_and_usage(
+    folders: Vec<String>,
+    imap: &mut Imap,
+) -> Result<Vec<(String, Vec<QuotaResource>)>> {
+    // IDEA: OPTIMIZATION:
```

Comment:
Quota is requested only once per day or something like that, no traffic can be saved with these optimizations. Optimizing it is not worth the added complexity and bugs as a result of synchronization failure. I suggest removing these todo comments.